### PR TITLE
Deprecated Boa lib replaced with latest boa_engine

### PR DESCRIPTION
**Pull request by samuchila**
Deprecated Boa 0.11 replaced with boa_engine 0.16. 
Changes:

- Value module is replaced with JsValue in boa_engine
- Parse implementation of Context is used to produce the StatementList for iteration.
- Statements.run() is disabled, hence replaced with context.eval()
- FunctionDecl.name() will return a Sym (Symbol data type). Hence interner of the context is used to convert to String.

### CODE REVIEW

File: api/src/apps/hooks.rs [hunk 1]
```diff
@@ -83,16 +83,17 @@ impl Apps {
             }
         };
 
-        let statements = match boa::parse(&hook_content, false) {
+        let mut context = Context::default();
+        let statements = match Context::default().parse(&hook_content) {
```
Comment (schrieveslaach) [inline api/src/apps/hooks.rs:87]: You can avoid an extra allocation here.

```suggestion
        let statements = match context.parse(&hook_content) {
```

File: api/src/apps/hooks.rs [hunk 2]
```diff
@@ -104,7 +105,7 @@ impl Apps {
             .items()
             .iter()
             .find_map(|node| match node {
-                Node::FunctionDecl(decl) if decl.name() == "deploymentHook" => Some(decl),
+                Node::FunctionDecl(decl) if context.interner().resolve_expect(decl.name()) == "deploymentHook" => Some(decl),
```
Comment (schrieveslaach) [inline api/src/apps/hooks.rs:108]: That might panic.

```suggestion
                Node::FunctionDecl(decl) if context.interner().resolve(decl.name()) == Some("deploymentHook") => Some(decl),
```

File: api/src/apps/hooks.rs [hunk 3]
```diff
@@ -117,7 +118,7 @@ impl Apps {
             .collect::<Vec<_>>();
 
         let js_configs = serde_json::to_value(js_configs).expect("Should be serializable");
-        let js_configs = Value::from_json(js_configs, context);
+        let js_configs = JsValue::from_json(&js_configs, context).unwrap();
```
Comment (schrieveslaach) [inline api/src/apps/hooks.rs:121]: We should handle the error case and maybe bubble that up.

**Review by schrieveslaach — CHANGES_REQUESTED** [review]
Overall the changes look good to me. Thanks for updating this dependency. I failed in the past.

There are some issues to be fixed. Additionally, you should run `cargo fmt` before commiting any changes.

Thanks in advance.

File: api/src/apps/hooks.rs (hunk 1 above)
Comment (samuchila) [reply api/src/apps/hooks.rs]: Extra allocation removed

File: api/src/apps/hooks.rs (hunk 2 above)
Comment (samuchila) [reply api/src/apps/hooks.rs]: Some() wrapper added

File: api/src/apps/hooks.rs (hunk 3 above)
Comment (samuchila) [reply api/src/apps/hooks.rs]: Error case is handled with below line
let js_configs =
            JsValue::from_json(&js_configs, &mut context).expect("Unable to read JSON value");

File: api/src/config/mod.rs [hunk 4]
```diff
@@ -196,9 +196,7 @@ impl Config {
     }
 
     pub fn hook(&self, hook_name: &str) -> Option<&PathBuf> {
-        self.hooks
-            .as_ref()
-            .and_then(|hooks| hooks.get(hook_name))
+        self.hooks.as_ref().and_then(|hooks| hooks.get(hook_name))
```
Comment (schrieveslaach) [inline api/src/config/mod.rs:199]: That's a change that is not related to PR's contribution. Please, revert.

File: api/src/apps/hooks.rs [hunk 5]
```diff
@@ -252,7 +260,7 @@ mod tests {
         let script = r#"
         function deploymentHook( appName, configs ) {
             if( appName === 'master' ) {
-                return configs.filter( service => service.name !== 'service-b' );
+                return configs.filter( service => service.name !== 'service-c' );
```
Comment (schrieveslaach) [inline api/src/apps/hooks.rs:263]: Why did you change this test? I would that the tests can be keep as they were before upgrading Boa

**Review by schrieveslaach — CHANGES_REQUESTED** [review]
There are some minor clean up tasks. Please, check them. And squash your commit into one commit. After that I'm going to merge and release the PR.

File: api/src/config/mod.rs (hunk 4 above)
Comment (samuchila) [reply api/src/config/mod.rs]: This has been reverted.

File: api/src/apps/hooks.rs (hunk 5 above)
Comment (samuchila) [reply api/src/apps/hooks.rs]: This has been reverted. Commits are squashed.

**Comment by schrieveslaach** [comment]
Thanks a lot for updating this component. @drik98 will be also happy about this update.
